aggregator/src/main/java/com/redhat/cloud/notifications/db/EmailAggregationResources.java

```diff
@@ -15,29 +17,55 @@
 public class EmailAggregationResources {
 
     @Inject
-    Session session;
+    EntityManager entityManager;
+
+    private static final String DEFAULT_ORG_ID = "NA";
+
+    public List<EmailAggregationKey> getApplicationsWithPendingAggregation(String orgIdToAggregate, LocalDateTime start, LocalDateTime end) {
+        String orgIdCriteria = "ea.org_id = :orgId AND acj.org_id IS NOT NULL";
+        if (DEFAULT_ORG_ID.equals(orgIdToAggregate)) {
+            orgIdCriteria = "acj.org_id IS NULL";
+        }
+        String query = "SELECT DISTINCT ea.org_id, ea.bundle, ea.application FROM email_aggregation ea "
+                + "LEFT JOIN aggregation_cronjob_parameter acj ON ea.org_id = acj.org_id "
```

Comment:
Is it really necessary to `JOIN` here? We already have the `org_id` value when the method is invoked and we do not select anything from `aggregation_cronjob_parameter` in the query.

Reply:
Please ignore that comment for now.